lib/src/component_declaration/redux_component.dart

```diff
@@ -0,0 +1,131 @@
+// Copyright 2016 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+library over_react.component_declaration.redux_component;
+
+import 'dart:async';
+
+import 'package:built_redux/built_redux.dart';
+import 'package:built_value/built_value.dart';
+import 'package:meta/meta.dart';
+import 'package:over_react/over_react.dart';
+
+/// Builds on top of [UiProps], adding typed props for [Store] in order to integrate with `built_redux`.
+///
+/// Use with the over_react transformer via the `@Props()` ([annotations.Props]) annotation.
+abstract class ReduxUiProps<V extends Built<V, B>, B extends Builder<V, B>,
+    A extends ReduxActions> extends UiProps {
+  String get _storePropKey => '${propKeyNamespace}store';
+
+  /// The prop defined by [StoresT].
+  ///
+  /// This object should either be an instance of [Store].
+  ///
+  /// __Instead of storing state within this component via `setState`, it is recommended that data be
+  /// pulled directly from these stores.__ This ensures that the data being used is always up to date
+  /// and leaves the state management logic to the stores.
+  Store<V, B, A> get store => props[_storePropKey];
+  set store(Store<V, B, A> value) => props[_storePropKey] = value;
+
+  A get actions => store.actions;
+}
+
+/// Builds on top of [UiComponent], adding `built_redux` integration.
+///
+/// * Redux components are responsible for rendering application views and turning
+///   user interactions and events into [Action]s.
+/// * Redux components can use data from one or many [Store] instances to define
+///   the resulting component.
+///
+/// Use with the over_react transformer via the `@Component()` ([annotations.Component]) annotation.
+///
+/// > Related: [ReduxUiStatefulComponent]
+abstract class ReduxUiComponent<V extends Built<V, B>, B extends Builder<V, B>,
```

Comment:
So this would require consumers to define V, B, and A on both the ReduxUiProps and ReduxUiComponent.

```dart
class MyProps extends ReduxUiProps<MyState, MyStateBuilder, MyActions> {}

class MyComponent extends ReduxUiComponent<MyState, MyStateBuilder, MyActions, MyProps> {}
```

I agree that it would be nicer to just have to put State, StateBuilder and Actions in the generic params of ReduxUiProps but it's just super verbose to have to have define them on the component as well

Reply:
Yea it is definitely not ideal. I would love to find a way to infer the types in one or the other. I'll noodle on it, but i'm struggling to find a way right now.

Reply:
Yeah, that's why I just have it passing the the `Store<V, B, A>` in my branch, I couldn't find a way around it 😢